src/prefect/backend/task_run.py

```diff
@@ -149,7 +187,10 @@ def _from_task_run_data(cls, task_run: dict) -> "TaskRunView":
         task_run = task_run.copy()  # Create a copy to avoid mutation
         task_run_id = task_run.pop("id")
         task_data = task_run.pop("task")
-        serialized_state = task_run.pop("serialized_state")
+
+        # The serialized state _could_ be null if the backend has not
+        # created it yet, this would typically be seen with mapped tasks
+        serialized_state = task_run.pop("serialized_state") or Scheduled().serialize()
```

Comment:
hmmmmmm are there any downsides to setting Scheduled here? feels like this could cause confusion if we get out of sync with the api

Reply:
This is only if the API has null data which only occurs if the task run was just created and hasn't been propagated yet